drivers/wdm/audio/legacy/wdmaud/deviface.c

```diff
@@ -49,10 +73,58 @@ DeviceInterfaceChangeCallback(
     IN PVOID NotificationStructure,
     IN PVOID Context)
 {
+#if 0
     DEVICE_INTERFACE_CHANGE_NOTIFICATION * Event = (DEVICE_INTERFACE_CHANGE_NOTIFICATION*)NotificationStructure;
+    PWDMAUD_DEVICE_EXTENSION DeviceExtension = (PWDMAUD_DEVICE_EXTENSION)Context;
+    SYSAUDIO_ENTRY * Entry;
+    ULONG Length;
+    NTSTATUS Status;
 
     DPRINT1("DeviceInterfaceChangeCallback called %p\n", Event);
-    DbgBreakPoint();
+
+    if (IsEqualGUIDAligned(&Event->Event,
+                           &GUID_DEVICE_INTERFACE_ARRIVAL))
+    {
+        Length = (Event->SymbolicLinkName->Length);
+        Entry = (SYSAUDIO_ENTRY*)AllocateItem(NonPagedPool, sizeof(SYSAUDIO_ENTRY) + (Length + 1)* sizeof(WCHAR));
+        if (!Entry)
+        {
+            return STATUS_INSUFFICIENT_RESOURCES;
+        }
+
+        Entry->SymbolicLink.Length = Length * sizeof(WCHAR);
+        Entry->SymbolicLink.MaximumLength = Length * sizeof(WCHAR);
+        Entry->SymbolicLink.Buffer = (LPWSTR) (Entry + 1);
+        RtlCopyMemory(Entry->SymbolicLink.Buffer, Event->SymbolicLinkName->Buffer, Length);
+
+        if (!IsEntryAlreadyInList(DeviceExtension, Entry)) {
```

Comment:
Could make this function just take the PCUNICODE_STRING as a parameter instead of the whole entry, then you could do this check before the allocation and just use `&Event->SymbolicLinkName`